frontend/src/routes/routes.tsx

```diff
@@ -0,0 +1,36 @@
+import { createBrowserRouter,redirect } from "react-router-dom";
+import { getTokens } from "../services/auth.service";
+import HomePage from "../pages/HomePage/HomePage";
+import LoginPage from "../pages/LoginPage/LoginPage";
+import NotFound from "../pages/NotFound/NotFound";
+import SignupPage from "../pages/SignupPage/SignupPage";
+import ErrorPage from "../pages/ErrorPage/ErrorPage";
+
+const authLoader = async () => {
+    const tokens = getTokens();
+    if (tokens.accessToken && tokens.refreshToken) {
```

Comment:
**Potential Fix**: Modify the authLoader function to handle cases where either the access token or the refresh token is missing. If either token is absent, redirect the user to the login page.

Reply:
**Problem**: There is no explicit error handling for failed route loading. If the `authLoader `function encounters an error while checking for tokens, the user may not be redirected appropriately.

**Fix**: Implement error handling in the loader function to catch any errors that occur during token retrieval. You can display an error page or redirect the user to a designated error route if token retrieval fails.